### PR TITLE
Autogrow currency

### PR DESCRIPTION
**Pull request by samkpower**
### Ch-ch-ch-changes
http://g.recordit.co/ALTCVyH4F9.gif

* Added an autogrow option to the ws-currency directive
* Added an autogrow-input attribute directive
* Added an autogrow input service for shared methods
* Added some really basic tests to prove I'm not breaking anything, but I wish I could test the keydown functionality. I tried it but couldn't get a keydown event to change the input value.
* Upgraded autoNumeric package to latest to take advantage of more sweet, sweet number formatting logic

### CODE REVIEW

File: src/form/ws-currency/ws-currency.js [hunk 1]
```diff
@@ -3,106 +3,147 @@
...
+    'ws.common.autogrowInput.service',
+  ];
 
+  function directive(autogrowInputService) {
+    const _autogrowInputService = autogrowInputService;
+    const { keyTypeForEventIs, getFontStyles, setInputWidth } = _autogrowInputService;
 
-  function link(scope, element, attr, ngModelCtrl) {
-    scope.options = scope.wsCurrency || {};
+    function link(scope, element, attr, ngModelCtrl) {
```
Comment (benjyblack) [inline src/form/ws-currency/ws-currency.js]: Why did you move the link function inside of the directive function? It might make it easier to read the diff if it were kept outside :P

Comment (samkpower) [inline src/form/ws-currency/ws-currency.js]: I ended up doing this when I injected the autogrowInputService - do you know if I can get an injected dependency from any of the arguments passed to the `link` function?

File: src/form/autogrow-input/autogrow-input.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const { keyTypeForEventIs, getFontStyles, setInputWidth } = _autogrowInputService;
+
+    return {
+      require: '?ngModel',
+      restrict: 'A',
+      link: link,
+    };
+
+    function link(scope, element, attrs, ngModel) {
+      scope.model = ngModel;
```
Comment (samkpower) [inline src/form/autogrow-input/autogrow-input.js]: Whoops! I'll remove this.

File: src/form/autogrow-input/autogrow-input.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  angular
+  .module('ws.common')
+  .directive('autogrowInput', directive);
+
+  directive.$inject = [
+    'ws.common.autogrowInput.service',
+  ];
+
+  function directive(autogrowInputService) {
+    const _autogrowInputService = autogrowInputService
```
Comment (samkpower) [inline src/form/autogrow-input/autogrow-input.js]: Note for pairing later: assigning this made the service available to all methods within the directive - if there is a better way though, I'd love to know!

**Comment by samkpower** [comment]
@benjyblack Finally got these to pass! Ended up removing the dist directory from AutoNumeric's gitignore and referenced that.
